### ISSUE TITLE
net/http: DetectContextType: recognize svg file as image/svg+xml

### ISSUE THREAD

**Issue by JadenTeng** [golang/go#15888]
**What version of Go are you using (`go version`)?**

`go version go1.6.2 darwin/amd64`

**What did you do?**
I want to upload svg file to aws s3. I use `http.DetectContentType(buffer)` to get the content type.
I really want the `http.DetectContentType(buffer)`; back `image/svg+xml` replace `text/xml; charset=utf-8`. Because the browser don't show the svg images when `<img src="s3/a-svg-file.svg">` if you give browser ContentType: `text/xml; charset=utf-8`.
I think it's more suit for web develop.

**What did you expect to see?**
 `image/svg+xml`

If the function out of `http.DetectContentType(buffer)` scope or you have suggestions for me. Welcome to reply.
Thank you very much and sorry for my bad English.

**Comment by JadenTeng** [comment]
I added extra logic to my code.
Get ContentType from Extension first.

```
fileType := mime.TypeByExtension(path.Ext(filePath))
if fileType == "" {
	fileType = http.DetectContentType(buffer)
}
```
It fix my problem, although don't feel nice.

If you feel don't need the feature. Please close this.
Sorry for the noise.

**Comment by ianlancetaylor** [comment]
Marking as 1.8 to decide whether or not to do it in 1.8.  It would help to know whether other systems, such as the file command, recognize SVG files, and how they do it.

**Comment by miguel-guedes** [comment]
@ianlancetaylor 

Landed here with the same problem.  The `file` command correctly produces `image/svg+xml` as an SVG file's mime type:

```sh
$ file -i /tmp/foo.svg
/tmp/foo.svg: image/svg+xml; charset=us-ascii
```

**Comment by bradfitz** [comment]
http.DetectContentType is documented to be an implementation of https://mimesniff.spec.whatwg.org/ and that mimesniff spec doesn't define svg.

And as you found, mime.TypeByExtension has ".svg" in its built-in table:

https://github.com/golang/go/blob/release-branch.go1.6/src/mime/type.go#L35

So I don't think there's anything to do here.

**Comment by odeke-em** [comment]
@bradfitz actually I reported an issue to whatwg https://github.com/whatwg/mimesniff/issues/7, when this issue was filed here so that perhaps if they could define its spec, they could work on it, add it to the spec and we could add it in here. The issue that I filed is still open, but from the looks of it doesn't seem like they'll be adding it in. If they add it in, I'll ping this bug and we can see what we can.
